pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -279,75 +236,51 @@ func (c *PruneController) sync(ctx context.Context, syncCtx factory.SyncContext)
 		return err
 	}
 
-	// update the config map
-	if err := c.updateRevisionStatusConfigMap(ctx, operatorStatus, syncCtx.Recorder()); err != nil {
-		return err
+	if operatorSpec.SucceededRevisionLimit == -1 && operatorSpec.FailedRevisionLimit == -1 {
+		return nil
 	}
 
-	failedLimit, succeededLimit := getRevisionLimits(operatorSpec)
-	excludedRevisions, err := c.excludedRevisionHistory(ctx, syncCtx.Recorder(), failedLimit, succeededLimit, defaultRevisionLimit)
+	// keep a number of revision before current, target, last failed and last available revisions
+	failedLimit, succeededLimit := revisionsToKeep(operatorSpec)
+	toKeep := c.revisionsToKeep(operatorStatus, failedLimit, succeededLimit)
+
+	// anything left?
+	cms, err := c.configMapGetter.ConfigMaps(c.targetNamespace).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
-	// if no IDs are excluded, then there is nothing to prune
-	if len(excludedRevisions) == 0 {
-		klog.Info("No excluded revisions to prune, skipping")
+	somethingFound := false
+	for _, cm := range cms.Items {
+		if !strings.HasPrefix(cm.Name, statusConfigMapName) {
+			continue
+		}
+		rev, err := strconv.Atoi(cm.Data["revision"])
+		if err != nil {
+			klog.Warningf("unexpected error converting revision to int: %+v", err)
+			continue
+		}
+		if !toKeep.Has(int32(rev)) {
+			somethingFound = true
+			break
+		}
+	}
+	if !somethingFound {
+		klog.Info("Nothing to prune")
 		return nil
 	}
 
 	errs := []error{}
-	if diskErr := c.pruneDiskResources(ctx, syncCtx.Recorder(), operatorStatus, excludedRevisions, excludedRevisions[len(excludedRevisions)-1]); diskErr != nil {
+	latest := toKeep.List()[len(toKeep)-1] // don't prune beyond this we are aware of
+	if diskErr := c.pruneDiskResources(ctx, syncCtx.Recorder(), operatorStatus, toKeep.List(), operatorStatus.LatestAvailableRevision); diskErr != nil {
```

Comment:
is `latest` != `operatorStatus.LatestAvailableRevision` ?

Reply:
no, it isn't. Fixed.